src/load/azext_load/data_plane/load_test/custom.py

```diff
@@ -302,6 +307,93 @@ def convert_to_jmx(
+        )
+    if existing_test_run.get("status") not in ["CANCELLED", "DONE"]:
+        raise InvalidArgumentValueError(
+            f"Test run with ID: {test_run_id} does not have a valid "
+            f"test run status {existing_test_run.get('status')}. "
+            "Valid test run status are: CANCELLED, DONE"
+        )
+    if existing_test_run.get("testRunStatistics", {}).get("Total") is None:
+        raise InvalidArgumentValueError(
+            f"Sampler statistics are not yet available for test run ID {test_run_id}. "
+            "Please try again later."
+        )
+    body = create_or_update_test_without_config(
+        test_id=test_id,
+        body=existing_test,
+        baseline_test_run_id=test_run_id,
+    )
+    response = test_client.create_or_update_test(test_id=test_id, body=body)
+    logger.debug("Set test run %s as baseline for test: %s", test_run_id, test_id)
+    return response.as_dict()
+
+
+def compare_to_baseline(
+    cmd,
+    load_test_resource,
+    test_id,
+    resource_group_name=None,
+    response_time_aggregate=AllowedTrendsResponseTimeAggregations.MEAN.value,
+):
+    logger.info("Showing test trends for test with test ID: %s", test_id)
+    test_client = get_admin_data_plane_client(cmd, load_test_resource, resource_group_name)
+    test_run_client = get_testrun_data_plane_client(cmd, load_test_resource, resource_group_name)
+    test = test_client.get_test(test_id)
+    if test.get("baselineTestRunId") is None:
+        raise InvalidArgumentValueError(
+            f"Test with ID: {test_id} does not have a baseline test run associated with it."
+        )
+    baseline_test_run_id = test.get("baselineTestRunId")
+    baseline_test_run = test_run_client.get_test_run(baseline_test_run_id)
+    all_test_runs = test_run_client.list_test_runs(test_id=test_id)
+    all_test_runs = [run.as_dict() for run in all_test_runs]
+    logger.debug("Total number of test runs: %s", len(all_test_runs))
+    count = 0
+    recent_test_runs = []
+    for run in all_test_runs:
+        if (
+            run.get("testRunId") != baseline_test_run_id
+            and count < 10  # Show only 10 most recent test runs
+            and run.get("status") in ["CANCELLED", "DONE"]
+        ):
+            recent_test_runs.append(run)
+            count += 1
+
+    logger.debug("Number of recent test runs: %s", len(recent_test_runs))
+    rows = [
+        generate_trends_row(baseline_test_run, response_time_aggregate=response_time_aggregate)
+    ]
+    for run in recent_test_runs:
+        rows.append(
+            generate_trends_row(run, response_time_aggregate=response_time_aggregate)
+        )
+    logger.debug("Retrieved test trends: %s", rows)
+    return rows
+
+
+def trends_output_transformer(result):
+    table = []
+    for row in result:
+        table.append(OrderedDict([(k, row.get(k)) for k in LoadTestTrendsKeys.ORDERED_HEADERS]))
```

Comment:
Just confirming, Wont `row.get(k)` error out? or will it return `None` ?

Reply:
it returns None, row[k] is what errors out with KeyError